File: 1094-car-pooling/1094-car-pooling.cpp
class Solution {
public:
    bool carPooling(vector<vector<int>>& trips, int capacity) {
        sort(trip.begin(), trips.end());
        vector<int> locations(1001);

        for (auto t : trips) {
            int passengers = t[0];
            int dep = t[1];
            int arr = t[2];
            locations[dep] += passengers;
            locations[arr] -= passengers;
        }

        int now = 0;
        for (auto p :locations) {
            now += p;
            if (now > capacity) return false;
        }
        return true;
    }
};

File: 1094-car-pooling/1094-car-pooling.py
class Solution:
    def carPooling(self, trips: List[List[int]], capacity: int) -> bool:
        heap = []
        locations = [0]*1001
        for passengers, dep, arr in trips:
            heap.append([(dep, passengers), (arr, -passengers)])
        heapify(heap)
        while capacity>=0 and heap:
            capacity += heappop(heap)[1][1]
            if capacity < 0:
                return False
        return True